echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+                                              on = [ 'stratum_num' ] ,
+                                              how = 'left' )
+    
+    # ---- Sum `weight_interp` over each stratum
+    proportions_unaged_weight_sex[ 'stratum_sex_weight_interp' ] = (
+        proportions_unaged_weight_sex.groupby( ['stratum_num' ] )[ 'weight_interp' ].transform( sum ) 
+    )
+
+    # ---- Calculate the normalized weights for each sex
+    proportions_unaged_weight_sex[ 'stratum_sex_weight_normalized' ] = (
+        proportions_unaged_weight_sex[ 'weight' ] *
+        proportions_unaged_weight_sex.weight_interp / proportions_unaged_weight_sex.stratum_sex_weight_interp
+    )
+
+    # ---- Sum the normalized stratum sex weight across each stratum
+    proportions_unaged_weight_sex[ 'proportions_weight_sex' ] = (
+        proportions_unaged_weight_sex.stratum_sex_weight_normalized /
+        proportions_unaged_weight_sex.weight_stratum_all
+    )
+
+    # ---- Sum 'proportions_weight_sex'
+    proportions_unaged_weight_sex[ 'proportions_weight_sex_total' ] = (
+        proportions_unaged_weight_sex.groupby( [ 'stratum_num' ] )[ 'proportions_weight_sex' ].transform( sum )
+    )
+
+    # ---- Calculate the final sex proportion
+    proportions_unaged_weight_sex[ 'proportion_weight_sex' ] = (
+        proportions_unaged_weight_sex.proportions_weight_sex / proportions_unaged_weight_sex.proportions_weight_sex_total
+    )
+
+    ### Remove unnecessary columns and return output
+    return proportions_unaged_weight_sex[ [ 'stratum_num' , 'sex' , 'proportion_weight_sex' ] ]
+
+def calculate_unaged_biomass( kriging_biomass_df: pd.DataFrame ,
+                              length_data: pd.DataFrame ,
+                              length_distribution: np.ndarray ,
+                              length_weight_df: pd.DataFrame ,
+                              aged_proportions: pd.DataFrame ):
+    """
+    Calculate the kriged biomass distributed over length and age for each sex and among
+    all fish (sexed)
+
+    Parameters
+    ----------
+    kriging_biomass_df: pd.DataFrame
+        Dataframe containing kriged biomass estimates
+    length_data: np.ndarray
+        Dataframe containing length data measured from unaged fish
+    length_distribution: pd.DataFrame
+        Dataframe containing the weight proportions for each length and age bin computed for all fish
+    length_weight_df: pd.DataFrame
+        Dataframe containing the modeled weights for each sex and length bin
+        fit from the length-weight regression (fit for all animals)
+    aged_proportions: pd.DataFrame
+        Dataframe containing the weight proportions of unaged fish within each stratum
+    """
+
+    ### Calculate number proportion
+    proportions_unaged_length = compute_unaged_number_proportions( length_data , length_distribution )
+
+    ### Calculate the normalized weight per unit length distribution (W_Ln_ALL in the original Matlab code)
```

Comment:
I would suggest changing this to avoid confusion because I don't think Chu's description was accurate:
```suggestion
    ### Calculate the weight proportion of unaged fish of each length bin (W_Ln_ALL in the original Matlab code)
```